addons/yatl/chat/chat.gd

```diff
@@ -1,163 +1,171 @@
-tool
-extends Node
-
-
 # Private imports
 
 const __PAL: Resource = preload("../pal/pal.gd")
+const __WebSocketConnection: Resource = __PAL.WebSocketConnection
 
 
-#######
-#
-# TODO: Update this code for new PAL abstraction
-#
-#######
+# Public imports
 
-
-#Custom class preloads
 const ChatMessage = preload("./chat_message.gd")
 
 
-#URL constants
+#Signals
+
+signal connected()
+signal login(success) # success: bool
+signal disconnected()
+signal connection_failed()
+signal channel_joined(channel) # channel: String
+signal channel_left(channel) # channel: String
+signal message_received(chat_message) # chat_message: ChatMessage
+signal whisper_received(chat_message) # chat_message: ChatMessage
+
+
+# URL constants
+
 const CHAT_WSS_URL: String = "wss://irc-ws.chat.twitch.tv:443"
 
 
-#Signals
-signal chat_connected(yatl_instance)
-signal chat_login(yatl_instance, success) # success: bool
-signal chat_disconnected(yatl_instance, clean) # clean: bool
-signal chat_connection_failed(yatl_instance)
-signal chat_joined(yatl_instance, channel) # channel: String
-signal chat_message_received(yatl_instance, chat_message) # chat_message: ChatMessage
-signal chat_whisper_received(yatl_instance, chat_message) # chat_message: ChatMessage
-signal chat_raw_data_received(yatl_instance, raw_data) # raw_data: String
-signal chat_ping_received(yatl_instance)
+# Public variables
+
+var reconnect: bool = false # Whether or not to try and reconnect automatically
+var message_queue_timeout: float = 0.33 # how long to wait after sending each message
```

Comment:
This could potentially live on the web socket connection instead.

Reply:
Agreed, then we wouldn't have to forward the base node's `_process()` call onto the chat class.